lib/configgin.rb

```diff
@@ -41,19 +46,35 @@ def generate_jobs(job_configs, templates)
         exit 1
       end
 
-      jobs[job] = Job.new(bosh_spec, kube_namespace, kube_client, kube_client_stateful_set)
+      jobs[job] = Job.new(
+        spec: bosh_spec,
+        namespace: kube_namespace,
+        client: kube_client,
+        client_stateful_set: kube_client_stateful_set,
+        self_name: @self_name
+      )
     end
     jobs
   end
 
-  def set_job_metadata(jobs)
+  # Set the exported properties and their digests, and return the digests
+  def patch_job_metadata(jobs)
+    digests = {}
     jobs.each do |name, job|
+      digest = property_digest(job.exported_properties)
       kube_client.patch_pod(
-        ENV['HOSTNAME'],
-        { metadata: { annotations: { :"skiff-exported-properties-#{name}" => job.exported_properties.to_json } } },
+        @self_name,
+        { metadata: {
```

Comment:
This formatting is confusing.

Reply:
Fixed in da315a15408eec9730f51d8d295c513d19f22eb2